runelite-client/src/main/java/net/runelite/client/plugins/blowpipetracker/BlowPipeTrackerPlugin.java

```diff
@@ -0,0 +1,367 @@
+	{
+		double dartRoll = random.nextDouble();
+		if (dartRoll <= attractorDefinition.getSavedChance())
+			return false;
+		else
+			return dartRoll > 1 - (attractorDefinition.getBreakOnImpactPercentage() + attractorDefinition.getDropToFloorChance());
+	}
+
+	private boolean shouldConsumeScales()
+	{
+		return random.nextDouble() <= 0.66;
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() != ChatMessageType.SERVER && event.getType() != ChatMessageType.FILTERED)
+			return;
+
+		String chatMsg = Text.removeTags(event.getMessage()); //remove color and linebreaks
+		if (chatMsg.contains(OUT_OF_DARTS))
+		{
+			if (config.showEmptyBlowpipeNotification())
+				notifier.notify("You have run out of darts!");
+
+			return;
+		}
+		if (chatMsg.contains(OUT_OF_SCALES))
+		{
+			if (config.showEmptyBlowpipeNotification())
+				notifier.notify("You have run out of scales!");
+
+			return;
+		}
+
+		//Extract dart quantity and type as well as number of scal
+		Matcher mComplete = DART_AND_SCALE_PATTERN.matcher(chatMsg);
+
+		List<String> matches = new ArrayList<>();
+		if (mComplete.find())
+		{
+			for (int i = 1; i <= mComplete.groupCount(); i++)
+			{
+				matches.add(mComplete.group(i));
+				log.debug("Match: " + mComplete.group(i));
+			}
+		}
+
+		switch (matches.size())
+		{
+			case 3:
+				dartId = getDartIdByName(matches.get(0));
+				dartsLeft = Integer.valueOf(matches.get(1).replace(",", ""));
+				scalesLeft = Integer.valueOf(matches.get(2).replace(",", ""));
+				parsedValuesFromText = true;
+				modifyCounters();
+				break;
+			default:
+				log.warn("Unknown match condition occurred.");
+				break;
+		}
+	}
+
+	@Subscribe
+	private void onConfigChanged(ConfigChanged event)
+	{
+		if (!event.getGroup().equals("blowpipetracker"))
+		{
+			return;
+		}
+
+		if (!config.showInfobox())
+		{
+			removeDartCounter();
+			removeScaleCounter();
+			removeCombinedCounter();
+		}
+		if (config.displayStyleMode() == DisplayStyleMode.COMBINED)
+		{
+			removeDartCounter();
+			removeScaleCounter();
+
+			if (dartsLeft >= 0 || scalesLeft >= 0)
+				clientThread.invokeLater(this::modifyCounters);
+		}
+		if (config.displayStyleMode() == DisplayStyleMode.INDIVIDUAL)
+		{
+			removeCombinedCounter();
+
+			if (dartsLeft >= 0 || scalesLeft >= 0)
+				clientThread.invokeLater(this::modifyCounters);
+		}
+	}
+
+	private void removeDartCounter()
+	{
+		if (dartCounter == null)
+			return;
+
+		infoBoxManager.removeInfoBox(dartCounter);
+		dartCounter = null;
+	}
+
+	private void removeScaleCounter()
+	{
+		if (scaleCounter == null)
+			return;
+
+		infoBoxManager.removeInfoBox(scaleCounter);
+		scaleCounter = null;
+	}
+
+	private void removeCombinedCounter()
+	{
+		if (combinedCounter == null)
+			return;
+
+		infoBoxManager.removeInfoBox(combinedCounter);
+		combinedCounter = null;
+	}
+
+	private int getDartIdByName(String dartName)
+	{
+		dartType = dartName;
```

Comment:
what is this

Reply:
This field (dartType) is used for setting the tooltip of the infobox

Reply:
cant you like use it at the place where you are setting infobox? This method is called getDartIdByName but still is mutating something mostly unrelated in class